docs/dev/README.md

```diff
@@ -0,0 +1,269 @@
+# Tekton Dashboard - Dev docs
+
+- Running tests
+- Troubleshooting
+
+This guide explains how to build, deploy and test the Tekton Dashboard. It covers the following topics:
+
+- [Before you begin](#before-you-begin)
+- [Pre-requisites](#pre-requisites)
+- [Checkout your fork](#checkout-your-fork)
+- [Build the frontend](#build-the-frontend)
+- [Build the backend](#build-the-backend)
+  - [Command line arguments](#command-line-arguments)
+- [Build YAML manifests](#build-yaml-manifests)
+- [Build and deploy with kustomize and ko](#build-and-deploy-with-kustomize-and-ko)
+  - [Build docker image for another GOARCH](#build-docker-image-for-another-goarch)
+- [Development server](#development-server)
+- [Run backend tests](#run-backend-tests)
+- [Run frontend tests](#run-backend-tests)
+  - [Unit tests](#unit-tests)
+  - [Linter](#linter)
+- [i18n](#i18n)
+- [Storybook](#storybook)
+- [Troubleshooting](#troubleshooting)
+- [Next steps](#next-steps)
+
+## Before you begin
+
+Many of the instructions here replicate what's in the [tektoncd/pipeline development guide](https://github.com/tektoncd/pipeline/blob/master/DEVELOPMENT.md), with a couple of caveats currently.
+
+- This project has not yet been tested with GKE
+
+## Pre-requisites
+
+In order to run the Tekton Dashboard, please make sure the requirements are met in [the install doc](../install.md) are met.
+
+You will also need the following tool in order to build the Dashboard locally and deploying it:
+1. [`go`](https://golang.org/doc/install): The language Tekton Dashboard is built in
+1. [`git`](https://help.github.com/articles/set-up-git/): For source control
+1. [Node.js & npm](https://nodejs.org/): For building and running the frontend locally. See `engines` in [package.json](../../package.json) for versions used. _Node.js 10.x is recommended_
+1. [`ko`](https://github.com/google/ko): For development. `ko` version v0.1 or higher is required for `dashboard` to work correctly
```

Comment:
soon to be 0.5.1 if we go ahead with https://github.com/tektoncd/dashboard/pull/1527

Reply:
i changed it to 0.5.0 until we switch to 0.5.1